### PR TITLE
Rest of models to toucan2 🎉

### PR DESCRIPTION
**Pull request by qnkhuat**
Epic https://github.com/metabase/metabase/issues/29231
- User
- Pulse, PulseCard, PulseChannel, PulseChannelRecipient
- Query, QueryCache, QueryExecution
- Metric, Segment, MetricImportantField
- ParameterCard
- NativeQuerySnippet
- DataMigrations
- FakeCard (testing)

### CODE REVIEW

File: test/metabase/models/user_test.clj [hunk 1]
```diff
@@ -463,16 +463,16 @@
...
+                             Pulse                 {pulse-id :id}         {}
+                             PulseChannel          {pulse-channel-id :id} {:pulse_id pulse-id}
+                             PulseChannelRecipient _ {:pulse_channel_id pulse-channel-id, :user_id user-id}]
       (letfn [(subscription-exists? []
                 (t2/exists? PulseChannelRecipient :pulse_channel_id pulse-channel-id, :user_id user-id))]
         (testing "Sanity check: subscription should exist"
           (is (subscription-exists?)))
         (testing "user is updated but not archived: don't delete the subscription"
-          (is (pos? (t2/update! User user-id {:is_active true})))
+          (is (pos? (t2/update! User user-id {:is_active true :first_name "New name"})))
```
Comment (metamben) [inline test/metabase/models/user_test.clj:475]: What's the reason for this change?

**Review by metamben — APPROVED** [review]
🎉

File: test/metabase/models/user_test.clj (hunk 1 above)
Comment (qnkhuat) [reply test/metabase/models/user_test.clj:475]: with the new change, `(t2/update!)` will returns 0 because it detects no change between the original version and the updates, so I added `first_name` to make sure the update will be executed.
